instrumentation/opentelemetry-api/opentelemetry-api-1.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/TracerTest.java

```diff
@@ -0,0 +1,329 @@
+public class TracerTest {
+
+  @RegisterExtension
+  private static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  @Test
+  @DisplayName("capture span, kind, attributes, and status")
+  public void captureSpanKindAttributesAndStatus() {
+    // When
+    Tracer tracer = GlobalOpenTelemetry.getTracer("test");
+    Span testSpan = tracer.spanBuilder("test").setSpanKind(PRODUCER).startSpan();
+    testSpan.setAttribute("string", "1");
+    testSpan.setAttribute("long", 2L);
+    testSpan.setAttribute("double", 3.0);
+    testSpan.setAttribute("boolean", true);
+    testSpan.setStatus(ERROR);
+    testSpan.end();
+
+    // Then
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("test")
+                        .hasKind(PRODUCER)
+                        .hasNoParent()
+                        .hasStatus(StatusData.error())
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(stringKey("string"), "1"),
+                            equalTo(longKey("long"), 2L),
+                            equalTo(doubleKey("double"), 3.0),
+                            equalTo(booleanKey("boolean"), true))));
+  }
+
+  @Test
+  @DisplayName("capture span with implicit parent using Tracer.withSpan()")
+  public void captureSpanWithImplicitParentUsingTracerWithSpan() {
+    // When
+    Tracer tracer = GlobalOpenTelemetry.getTracer("test");
+    Span parentSpan = tracer.spanBuilder("parent").startSpan();
+    Scope parentScope = Context.current().with(parentSpan).makeCurrent();
+
+    Span testSpan = tracer.spanBuilder("test").startSpan();
+    testSpan.end();
+
+    parentSpan.end();
+    parentScope.close();
+
+    // Then
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span0 -> span0.hasName("parent").hasNoParent().hasAttributesSatisfyingExactly(),
+                span1 ->
+                    span1
+                        .hasName("test")
+                        .hasParent(trace.getSpan(0))
+                        .hasAttributesSatisfyingExactly()));
+  }
+
+  @Test
+  @DisplayName("capture span with implicit parent using makeCurrent")
+  public void captureSpanWithImplicitParentUsingMakeCurrent() {
+    // When
+    Tracer tracer = GlobalOpenTelemetry.getTracer("test");
+    Span parentSpan = tracer.spanBuilder("parent").startSpan();
+    Scope parentScope = parentSpan.makeCurrent();
+
+    Span testSpan = tracer.spanBuilder("test").startSpan();
+    testSpan.end();
+
+    parentSpan.end();
+    parentScope.close();
+
+    // Then
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span0 -> span0.hasName("parent").hasNoParent(),
+                span1 -> span1.hasName("test").hasParent(trace.getSpan(0))));
```

Comment:
```suggestion
                span -> span.hasName("test").hasParent(trace.getSpan(0)).hasTotalAttributeCount(0)));
```

this same suggestion applies for the for each of the spans that have the `attributes {}` in the groovy version